salt/files/orchestrate.sls

```diff
@@ -1,36 +1,116 @@
 {%- from "salt/map.jinja" import master with context %}
 {%- if master.enabled %}
 
-{{ formula_dict }}
-
 {%- for environment_name, environment in master.get('environment', {}).iteritems() %}
 
 {%- if master.base_environment == environment_name %}
 
-{%- set formula_dict = environment.get('formula', {}) %}
-{%- set new_formula_dict = {} %}
+{%- set sorted_priorities = priorities|dictsort(false, 'value') %}
 
-{%- for formula_name, formula in formula_dict.iteritems() %}
+{#- Set debug = True to log simple state result - Fail/True - to /var/log/salt/ on both master and respective minions #}
+{%- set debug = False %}
 
-{%- set _tmp = new_formula_dict.update({formula_name: formula.get('orchestrate_order', 100)}) %}
+{#- Uncomment to print parsed metadata from formula_name/meta/salt.yml to output file %}
 
-{%- endfor %}
+PARSED METADATA: 
+
+---------------
+
+Passed from salt/master/orchestrate.sls
+---
+
+priorities: {{ priorities }}
+
+sorted_priorities: {{ sorted_priorities }}
+
+args: {{ args }}
+
+---------------
+
+#}
+
+{%- for state in sorted_priorities %}
+
+{%- set formula = state.0.split('.') %}
 
-{%- set sorted_formula_list = new_formula_dict|dictsort(false, 'value') %}
-	
-{%- for formula in sorted_formula_list %}
+{%- if salt['file.directory_exists']('/srv/salt/env/'+environment_name+'/'+formula.0+'/orchestrate') and formula|length > 1 and salt['file.file_exists']('/srv/salt/env/'+environment_name+'/'+formula.0+'/orchestrate/'+formula.1+'.sls') %}
 
-{%- if salt['file.file_exists']('/srv/salt/env/'+environment_name+'/'+formula.0+'/orchestrate.sls') %}
+{{ salt['cmd.run']('cat /srv/salt/env/'+environment_name+'/'+formula.0+'/orchestrate/'+formula.1+'.sls') }}
 
-{{ salt['cmd.run']('cat /srv/salt/env/'+environment_name+'/'+formula.0+'/orchestrate.sls') }}
+{%- else %}
+
+{%- if args[ state.0 ] is defined %}
+
+{{ state.0 }}:
+  salt.state:
+    - tgt: '{{ state.0|replace(".", ":") }}'
+    - tgt_type: pillar
```

Comment:
Compound allows you to match minion using multiple methods (grain, pillar, minion ID etc. all together), there is no other method I need to (or even can) use, most usable grains items (role for instance) are only generated after salt state ran on the minion, so pillar matching felt more suitable (its always available) and minion IDs can be different on various deployments